### PR TITLE
Iqra progress page 2

### PR DESCRIPTION
**Pull request by iqra-javed**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

- Added fetch and delete progress functionality on the back end and front end.
- Also modified progress reducer for ADD_PROGRESS_SUCCESS action. Now when a progress is added, it appears on the page without having to force a reload.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Tested on local host and mlab

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

File: client/src/components/Progress.js [hunk 1]
```diff
@@ -55,81 +57,105 @@ class Progress extends Component {
...
+    progressRecords: state.progress.progressRecords
+  };
+};
 
 Progress.propTypes = {
-    addProgress: PropTypes.func,
-}
-
-
+  addProgress: PropTypes.func,
```
Comment (rusticlion) [inline client/src/components/Progress.js:154]: I'd like if you could talk about PropTypes tomorrow, so we can all start using it.

File: client/src/less/progress.less [hunk 2]
```diff
@@ -0,0 +1,51 @@
+.container {
```
Comment (rusticlion) [inline client/src/less/progress.less:1]: You might be interested in reviewing [some of the powerful features LESS gives you](http://lesscss.org/features/#variables-feature), but keep in mind that LESS is backwards compatible, so all your CSS knowledge applies. Now is the time for experimentation!

File: client/src/reducers/progress.js [hunk 3]
```diff
@@ -1,28 +1,55 @@
 import * as Actions from "../actions/actionDefinitions";
```
Comment (rusticlion) [inline client/src/reducers/progress.js:1]: Good use of a separate reducer. Come to think of it, it probably makes sense to have roughly one per page.

File: server/api/models/Progress.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const mongoose = require("mongoose");
```
Comment (rusticlion) [inline server/api/models/Progress.js:1]: Looking at this, I feel like maybe we defined the "categories" for measuring progress too strictly. Would it make more sense to have the user define their own categories?

**Review by rusticlion — APPROVED** [review]
Looks good, we're starting to get features!

**Review by emukupa — APPROVED** [review]
The code looks good 👍

**Pull request merged**